tests/test_mad.py

```diff
@@ -0,0 +1,577 @@
+# -*- coding: utf-8 -*-
+"""MADUpdate.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/10W6-JLI26FsVsrpolWGG9Gb7YjDoS3hy
+"""
+
+import numpy as np
+import torch
+from torchvision import transforms
+import torch.nn as nn
+import torch.nn.functional as F
+import math
+import functools
+
+
+
+def abs(x):
+    return torch.sqrt(x[:,:,:,:,0]**2+x[:,:,:,:,1]**2+1e-12)
+
+def real(x):
+    return x[:,:,:,:,0]
+
+def imag(x):
+    return x[:,:,:,:,1]
+
+def roll_n(X, axis, n):
+    f_idx = tuple(slice(None, None, None) if i != axis else slice(0, n, None) for i in range(X.dim()))
+    b_idx = tuple(slice(None, None, None) if i != axis else slice(n, None, None) for i in range(X.dim()))
+    front = X[f_idx]
+    back = X[b_idx]
+    return torch.cat([back, front], axis)
+
+def batch_fftshift2d(x):
+    real, imag = torch.unbind(x, -1)
+    for dim in range(1, len(real.size())):
+        n_shift = real.size(dim)//2
+        if real.size(dim) % 2 != 0:
+            n_shift += 1  # for odd-sized images
+        real = roll_n(real, axis=dim, n=n_shift)
+        imag = roll_n(imag, axis=dim, n=n_shift)
+    return torch.stack((real, imag), -1)  # last dim=2 (real&imag)
+
+def batch_ifftshift2d(x):
+    real, imag = torch.unbind(x, -1)
+    for dim in range(len(real.size()) - 1, 0, -1):
+        real = roll_n(real, axis=dim, n=real.size(dim)//2)
+        imag = roll_n(imag, axis=dim, n=imag.size(dim)//2)
+    return torch.stack((real, imag), -1)  # last dim=2 (real&imag)
+
+def preprocess_lab(lab):
+		L_chan, a_chan, b_chan =torch.unbind(lab,dim=2)
+		# L_chan: black and white with input range [0, 100]
+		# a_chan/b_chan: color channels with input range ~[-110, 110], not exact
+		# [0, 100] => [-1, 1],  ~[-110, 110] => [-1, 1]
+		return [L_chan / 50.0 - 1.0, a_chan / 110.0, b_chan / 110.0]
+
+def deprocess_lab(L_chan, a_chan, b_chan):
+		#TODO This is axis=3 instead of axis=2 when deprocessing batch of images 
+			   # ( we process individual images but deprocess batches)
+		#return tf.stack([(L_chan + 1) / 2 * 100, a_chan * 110, b_chan * 110], axis=3)
+		return torch.stack([(L_chan + 1) / 2.0 * 100.0, a_chan * 110.0, b_chan * 110.0], dim=2)
+
+def rgb_to_lab(srgb):
```

Comment:
We have colour conversion functions in `piq.functonal`.  Check it out and remove this functions along with `preprocess_lab` and `deprocess_lab`